src/routes/accounts/accounts.service.ts

```diff
@@ -69,4 +94,23 @@ export class AccountsService {
       domainDataType.is_active,
     );
   }
+
+  private mapDataSetting(
+    dataTypes: DomainAccountDataType[],
+    domainAccountDataSetting: DomainAccountDataSetting,
+  ): AccountDataSetting {
+    const dataType = dataTypes.find(
+      (dt) => dt.id === domainAccountDataSetting.account_data_type_id,
+    );
+
+    if (!dataType) {
+      throw new Error('Data type not found');
+    }
+
+    return {
+      dataTypeName: dataType.name,
+      dataTypeDescription: dataType.description,
```

Comment:
As before regarding the inclusion of an `id` and not prefixing with "dataType-".

Reply:
Thanks! Will refactor this based on the outcome of [the other comment](https://github.com/safe-global/safe-client-gateway/pull/1715#discussion_r1668444149) 🙂